erpnext/templates/includes/integrations/authorizenet_checkout.js

```diff
@@ -1,87 +1,308 @@
-	let expirationDate = expirationYear.concat("-").concat(expirationMonth);
-	let cardCode = document.getElementById('card-code').value;
-	let isValidCard = frappe.cardValidator.number(cardNumber);
-
-	if (!cardHolderName) {
-		frappe.throw(__("Card Holder Name is mandatory."));
-	}
+/**
+ * On Success Callback
+ * @typedef {function(object):Promise<void>} onCardProcessSuccess
+ * @callback onCardProcessSuccess
+ * @param {ProcessStatus} status The returned status object from the server
+ */
 
-	if (!cardHolderEmail) {
-		frappe.throw(__("Card Holder Email is mandatory."));
-	}
-	
-	if (!validate_email(cardHolderEmail)) {
-		frappe.throw(__("Card Holder Email is invalid."));
-	}
+/**
+ * On Failure Callback
+ * @typedef {function(object):Promise<void>} onCardProcessFail
+ * @callback onCardProcessFail
+ * @param {ProcessStatus} status returned status object from the server
+ * @param {Exception} err If one occurs, the exception instance.
+ */
 
-	if (!validate_email(cardHolderEmail)) {
-		frappe.throw(__("Card Holder Email is invalid."));
-	}
+/**
+* @typedef {Object} ProcessStatus
+* @property {string} status Status enum. Possible values: "Completed", "NotFound", "Queued", "Failed".
+* @property {string} type On a failed request. The type is one of: "Validation", "HardError", "Timeout", "Unhandled"
+* @property {string} description A reason message for the status returned.
+* @property {int} wait An amount of time the client should wait before issuing 
+*                      another status check request in milliseconds.
+*/
+
+frappe.ready(function() {
+	const data = context.replace(/'/g, '"');
+
+	/**
+	 * Fetches payment request status information
+	 * @param {*} data Order information data structure. See authorizenet_settings.py
+	 * @returns Promise<void>
+	 */
+	const fetch_status = async function(data) {
+		const r = await frappe.call({
+			method: "erpnext.erpnext_integrations.doctype.authorizenet_settings.authorizenet_settings.get_status",
+			freeze: true,
+			args: {
+				"data": data
+			}
+		});
 
-	if (!isValidCard.isPotentiallyValid) {
-		frappe.throw(__("Card Number is Invalid."));
+		return r.message;
 	}
 
-	if(cardNumber.length < 13 || cardNumber.length > 16){
-		frappe.throw(__("Card Number length should be between 13 and 16 characters"));
+	/**
+	 * Starts status request and UI updates to handle PR status check.
+	 * @param {object} data Order information data structure. See authorizenet_settings.py
+	 * @param {string} label Status label to display when processing status
+	 * @param {onCardProcessSuccess} on_success a Success function callback. 
+	 * @param {onCardProcessFail} on_fail A Failure function callback
+	 * @param {int} counter A counter which is incremented internally as timeouts occur
+	 * @param {int} maxTimeouts Maximum number of timeouts before a hard error
+	 * @returns Promise<void>
+	 */
+	const check_status = async function(data, label, on_success, on_fail, counter, maxTimeouts) {
+		if ( !maxTimeouts) {
```

Comment:
```suggestion
		if (!maxTimeouts) {
```
Can we please use this syntax/formatting style everywhere in this file just to keep it consistent with frappe styling?